move/interchain_token_service/sources/versioned/interchain_token_service_v0.move

```diff
@@ -509,6 +511,8 @@ module interchain_token_service::interchain_token_service_v0 {
         treasury_cap_reclaimer: TreasuryCapReclaimer<T>,
         token_id: TokenId,
     ): TreasuryCap<T> {
+        assert!(token_id == treasury_cap_reclaimer.token_id(), EWrongTreasuryCapReclaimer);
```

Comment:
`token_id` should be retrieved from the reclaimer cap instead of being passed separately